src/lib/math.js

```diff
@@ -314,6 +314,55 @@ const $builtinmodule = function (name) {
         throw new Sk.builtin.NotImplementedError("math.isqrt() is not yet implemented in Skulpt");
     };
 
+    function lcm() {
+        const nargs = arguments.length;
+
+        // lcm() without arguments returns 1
+        if (nargs === 0) return new Sk.builtin.int_(1);
+
+        let result = arguments[0];
+        Sk.builtin.pyCheckType("*integers", "integer", Sk.builtin.checkInt(result));
+        result = Sk.builtin.asnum$(result);
+
+        if (nargs === 1) {
+            if (typeof result === "number") {
+                return new Sk.builtin.int_(Math.abs(result));
+            } else {
+                return new Sk.builtin.int_(result);
+            }
+        }
+
+        let i;
+        for (i = 1; i < nargs; ++i) {
+          const argument = arguments[i];
+
+          Sk.builtin.pyCheckType(
+            "*integers",
+            "integer",
+            Sk.builtin.checkInt(argument)
+          );
+
+          // If any of the arguments is zero, then the returned value is 0
+          if (argument === 0) return 0;
+
+          let _argument = Sk.builtin.asnum$(argument);
+
+          if (typeof result === "number" && typeof _argument === "number") {
+            result = (result / gcd(result, _argument)) * _argument;
```

Comment:
for the `gcd` function: i'd consider extracting the logic that just does the javascript part into a separate function.
Otherwise we end up going through all the type checks and conversions to javascript again.

Reply:
Something like this?

```js
    function _gcd_internal(_a, _b) {
        function _gcd(a, b) {
            if (b == 0) {
                return a;
            }
            return _gcd(b, a % b);
        }
        
        function _biggcd(a, b) {
            if (JSBI.equal(b, JSBI.__ZERO)) {
                return a;
            }
            return _biggcd(b, JSBI.remainder(a, b));
        }

        let res;
        if (typeof _a === "number" && typeof _b === "number") {
            _a = Math.abs(_a);
            _b = Math.abs(_b);
            res = _gcd(_a, _b);
            res = res < 0 ? -res : res; // python only returns positive gcds
            return new Sk.builtin.int_(res);
        } else {
            _a = JSBI.BigInt(_a);
            _b = JSBI.BigInt(_b);
            res = _biggcd(_a, _b);
            if (JSBI.lessThan(res, JSBI.__ZERO)) {
                res = JSBI.multiply(res, JSBI.BigInt(-1));
            } 
            return new Sk.builtin.int_(res.toString()); // int will convert strings
        }
    };

    function gcd(a, b) {
        // non ints not allowed in python 3.7.x
        Sk.builtin.pyCheckType("a", "integer", Sk.builtin.checkInt(a));
        Sk.builtin.pyCheckType("b", "integer", Sk.builtin.checkInt(b));

        let _a = Sk.builtin.asnum$(a);
        let _b = Sk.builtin.asnum$(b);

        return _gcd_internal(_a, _b);
    };
```

And then use _gcd_internal in lcm function.

Reply:
Yes although - I don't think you want the internal version to return a python object.

i.e. return the raw res from `_gcd_internal` and the final line of `gcd` should be something like 

```javascript
const res = _gcd_internal(_a, _b);
// res is positive
if (typeof res === "number") {
    return new Sk.builtin.int_(res);
} else {
    return new Sk.builtin.int_(res.toString());
    // this is a bit of a hack - we're being lazy and passing a string to int_
    // the int constructor will then decide if it should be a number or a BigInt depending on the size.
    // really we should do:
    // return  JSBI.lessThanOrEqual(res, JSBI.BigInt(Number.MAX_SAFE_INTEGER)) ? new Sk.builtin.int_(Number(res)) : new Sk.builtin.int_(res);
}
```